core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -269,19 +269,22 @@ class ReplicaFetcherThread(name: String,
    * Truncate the log for each partition's epoch based on leader's returned epoch and offset.
    * The logic for finding the truncation offset is implemented in AbstractFetcherThread.getOffsetTruncationState
    */
-  override def truncate(tp: TopicPartition, offsetTruncationState: OffsetTruncationState): Unit = {
-    val replica = replicaMgr.localReplicaOrException(tp)
-    val partition = replicaMgr.getPartition(tp).get
-    partition.truncateTo(offsetTruncationState.offset, isFuture = false)
-
-    if (offsetTruncationState.offset < replica.highWatermark.messageOffset)
-      warn(s"Truncating $tp to offset ${offsetTruncationState.offset} below high watermark " +
-        s"${replica.highWatermark.messageOffset}")
-
-    // mark the future replica for truncation only when we do last truncation
-    if (offsetTruncationState.truncationCompleted)
-      replicaMgr.replicaAlterLogDirsManager.markPartitionsForTruncation(brokerConfig.brokerId, tp,
-        offsetTruncationState.offset)
+  override def truncate(tp: TopicPartition, offsetTruncationState: OffsetTruncationState,
+                        offsetTruncationStates: Option[Map[TopicPartition, OffsetTruncationState]] = None): Unit = {
```

Comment:
Could we make offsetTruncationStates non-optional and get rid of tp and offsetTruncationState?